spinoso-time/src/time/tzrs/math.rs

```diff
@@ -0,0 +1,491 @@
+
+    fn add(self, seconds: u64) -> Self::Output {
+        let duration = Duration::from_secs(seconds);
+        self + duration
+    }
+}
+
+impl Add<f32> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: f32) -> Self::Output {
+        self + f64::from(seconds)
+    }
+}
+
+impl Add<f64> for Time {
+    type Output = Self;
+
+    fn add(self, seconds: f64) -> Self::Output {
+        if seconds.is_sign_positive() {
+            self + Duration::from_secs_f64(seconds)
+        } else {
+            let seconds = -seconds;
+            self - Duration::from_secs_f64(seconds)
+        }
+    }
+}
+
+impl Sub<Duration> for Time {
+    type Output = Self;
+
+    fn sub(self, duration: Duration) -> Self::Output {
+        let unix_time = self.inner.unix_time();
+        let nanoseconds = self.inner.nanoseconds();
+        let offset = self.offset;
+
+        let duration_seconds = i64::try_from(duration.as_secs()).expect("Duration too large");
+        let duration_subsecs = duration.subsec_nanos();
+
+        let mut seconds = unix_time.checked_sub(duration_seconds).expect("Duration too large");
+        let nanoseconds = if let Some(nanos) = nanoseconds.checked_sub(duration_subsecs) {
+            nanos
+        } else {
+            seconds -= 1;
+            nanoseconds + NANOS_IN_SECOND - duration_subsecs
+        };
+
+        Self::Output::with_timespec_and_offset(seconds, nanoseconds, offset)
+    }
+}
+
+impl Sub<i8> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: i8) -> Self::Output {
+        self - i64::from(seconds)
+    }
+}
+
+impl Sub<u8> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: u8) -> Self::Output {
+        self - u64::from(seconds)
+    }
+}
+
+impl Sub<i16> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: i16) -> Self::Output {
+        self - i64::from(seconds)
+    }
+}
+
+impl Sub<u16> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: u16) -> Self::Output {
+        self - u64::from(seconds)
+    }
+}
+
+impl Sub<i32> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: i32) -> Self::Output {
+        self - i64::from(seconds)
+    }
+}
+
+impl Sub<u32> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: u32) -> Self::Output {
+        self - u64::from(seconds)
+    }
+}
+
+impl Sub<i64> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: i64) -> Self::Output {
+        self + -seconds
+    }
+}
+
+impl Sub<u64> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: u64) -> Self::Output {
+        let duration = Duration::from_secs(seconds);
+        self - duration
+    }
+}
+
+impl Sub<f32> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: f32) -> Self::Output {
+        self - f64::from(seconds)
+    }
+}
+
+impl Sub<f64> for Time {
+    type Output = Self;
+
+    fn sub(self, seconds: f64) -> Self::Output {
+        self + -seconds
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I love all these tests 😍

Reply:
I did borrow half of them from your chrono crate :sweat_smile: (The rounding ones etc are all new though :smile:)